### PR TITLE
add e2e test case user reset mfa

### PR DESCRIPTION
**Pull request by M-Buntoro**
### Summary

Add end to end test case for `mmctl user resetmfa` 

### Ticket Link
https://mattermost.atlassian.net/browse/MM-29091

Fixes https://github.com/mattermost/mattermost-server/issues/15628

### CODE REVIEW

File: commands/user_e2e_test.go [hunk 1]
```diff
@@ -136,3 +138,77 @@ func (s *MmctlE2ETestSuite) TestSearchUserCmd() {
...
 	})
 }
+
+func (s *MmctlE2ETestSuite) TestResetUserMfaCmd() {
+	s.SetupTestHelper().InitBasic()
+
+	user, appErr := s.th.App.CreateUser(&model.User{Email: s.th.GenerateTestEmail(), Username: model.NewId(), Password: model.NewId(), MfaActive: true, MfaSecret: "secret"})
+	s.Require().Nil(appErr)
+
+	s.Run("Reset user mfa", func() {
```
Comment (isacikgoz) [inline commands/user_e2e_test.go:148]:
```suggestion
	s.RunForSystemAdminAndLocal("Reset user mfa", func(c client.Client) {
```

File: commands/user_e2e_test.go [hunk 2]
```diff
@@ -136,3 +138,77 @@ func (s *MmctlE2ETestSuite) TestSearchUserCmd() {
...
+		defer func() {
+			s.th.App.UpdateConfig(func(cfg *model.Config) { *cfg.ServiceSettings.EnableMultifactorAuthentication = *previousVal })
+		}()
+
+		s.th.AddPermissionToRole(model.PERMISSION_EDIT_OTHER_USERS.Id, model.SYSTEM_ADMIN_ROLE_ID)
+		defer func() {
+			s.th.RemovePermissionFromRole(model.PERMISSION_EDIT_OTHER_USERS.Id, model.SYSTEM_ADMIN_ROLE_ID)
+		}()
+
+		err := resetUserMfaCmdF(s.th.SystemAdminClient, &cobra.Command{}, []string{user.Email})
```
Comment (isacikgoz) [inline commands/user_e2e_test.go:162]:
```suggestion
		err := resetUserMfaCmdF(c, &cobra.Command{}, []string{user.Email})
```

File: commands/user_e2e_test.go [hunk 3]
```diff
@@ -136,3 +138,77 @@ func (s *MmctlE2ETestSuite) TestSearchUserCmd() {
...
+		previousVal := s.th.App.Config().ServiceSettings.EnableMultifactorAuthentication
+		s.th.App.UpdateConfig(func(cfg *model.Config) { *cfg.ServiceSettings.EnableMultifactorAuthentication = true })
+		defer func() {
+			s.th.App.UpdateConfig(func(cfg *model.Config) { *cfg.ServiceSettings.EnableMultifactorAuthentication = *previousVal })
+		}()
+
+		s.th.AddPermissionToRole(model.PERMISSION_EDIT_OTHER_USERS.Id, model.SYSTEM_ADMIN_ROLE_ID)
+		defer func() {
+			s.th.RemovePermissionFromRole(model.PERMISSION_EDIT_OTHER_USERS.Id, model.SYSTEM_ADMIN_ROLE_ID)
+		}()
```
Comment (isacikgoz) [inline commands/user_e2e_test.go:160]: I don't think `SYTEM_ADMIN` needs an extra permission to do things :)
```suggestion
```

**Review by isacikgoz — CHANGES_REQUESTED** [review]
Hey @M-Buntoro, thanks again for the contribution!! 🎉 I have some change requests to improve the readability and prevent some unnecessary executions.

File: commands/user_e2e_test.go (hunk 3 above)
Comment (M-Buntoro) [reply commands/user_e2e_test.go:160]: Hmm, I see, seems my bad on that one, still learning about flow in mattermost,
It seems to work well with local and system_admin, let me adjust to that 

thanks @isacikgoz :+1:

File: commands/user_e2e_test.go [hunk 4]
```diff
@@ -136,3 +138,67 @@ func (s *MmctlE2ETestSuite) TestSearchUserCmd() {
...
+	s.Require().Nil(appErr)
+
+	s.RunForSystemAdminAndLocal("Reset user mfa", func(c client.Client) {
+		printer.Clean()
+
+		previousVal := s.th.App.Config().ServiceSettings.EnableMultifactorAuthentication
+		s.th.App.UpdateConfig(func(cfg *model.Config) { *cfg.ServiceSettings.EnableMultifactorAuthentication = true })
+		defer func() {
+			s.th.App.UpdateConfig(func(cfg *model.Config) { *cfg.ServiceSettings.EnableMultifactorAuthentication = *previousVal })
+		}()
```
Comment (mgdelacroix) [inline commands/user_e2e_test.go:155]: nit: the wrapping `func` is not needed, same in the rest of the `defer` oneliners

```suggestion
		defer s.th.App.UpdateConfig(func(cfg *model.Config) { *cfg.ServiceSettings.EnableMultifactorAuthentication = *previousVal })
```

File: commands/user_e2e_test.go [hunk 5]
```diff
@@ -136,3 +138,67 @@ func (s *MmctlE2ETestSuite) TestSearchUserCmd() {
...
+		printer.Clean()
+
+		previousVal := s.th.App.Config().ServiceSettings.EnableMultifactorAuthentication
+		s.th.App.UpdateConfig(func(cfg *model.Config) { *cfg.ServiceSettings.EnableMultifactorAuthentication = false })
+		defer func() {
+			s.th.App.UpdateConfig(func(cfg *model.Config) { *cfg.ServiceSettings.EnableMultifactorAuthentication = *previousVal })
+		}()
+
+		userMfaInactive, appErr := s.th.App.CreateUser(&model.User{Email: s.th.GenerateTestEmail(), Username: model.NewId(), Password: model.NewId(), MfaActive: false})
+		s.Require().Nil(appErr)
```
Comment (mgdelacroix) [inline commands/user_e2e_test.go:178]: this user can be created before the first test case so it's not created twice as part of the `RunForSystemAdminAndLocal` helper loop

**Review by mgdelacroix — APPROVED** [review]
Really nice work @M-Buntoro!! Thanks a lot for the contribution! :rocket: 

Added a couple of minor suggestions, none blocking the PR

**Comment by mgdelacroix** [comment]
Please @M-Buntoro take a look at the linter running `make check` locally to get the CI green :slightly_smiling_face:

File: commands/user_e2e_test.go (hunk 5 above)
Comment (M-Buntoro) [reply commands/user_e2e_test.go:176]: agree, let me move that code

**Comment by M-Buntoro** [comment]
> Please @M-Buntoro take a look at the linter running `make check` locally to get the CI green slightly_smiling_face

on it :ok_hand:

**Review by isacikgoz — APPROVED** [review]
LGTM, thanks for the contribution @M-Buntoro 🎉

**Review by lindy65 — APPROVED** [review]
Thanks @M-Buntoro @mgdelacroix - looks good to me 👍

**Comment by hanzei** [comment]
Sorry it took so long to merge the PR. Thanks @M-Buntoro :+1:

**Pull request merged**